File: check_github.py
import urllib.request
import json

url = "https://api.github.com/repos/Dev-By-Varshith/vidhi-trading-platform/actions/runs"
req = urllib.request.Request(url)
req.add_header('User-Agent', 'Mozilla/5.0')
with urllib.request.urlopen(req) as response:
    data = json.loads(response.read().decode())
    runs = data.get('workflow_runs', [])[:5]
    for r in runs:
        print(f"Run {r['id']}: {r['name']} | Status: {r['status']} | Conclusion: {r['conclusion']} | Created: {r['created_at']}")

File: check_github.js
async function run() {
  const res = await fetch('https://api.github.com/repos/Dev-By-Varshith/vidhi-trading-platform/actions/runs?per_page=1');
  const data = await res.json();
  const runId = data.workflow_runs[0].id;
  console.log("Run ID:", runId);
  const jobsRes = await fetch(`https://api.github.com/repos/Dev-By-Varshith/vidhi-trading-platform/actions/runs/${runId}/jobs`);
  const jobsData = await jobsRes.json();
  for (const job of jobsData.jobs) {
    if (job.conclusion === 'failure') {
      console.log("Failed Job:", job.name);
      for (const step of job.steps) {
        if (step.conclusion === 'failure') {
          console.log("Failed Step:", step.name);
        }
      }
    }
  }
}
run();
